build-tools/packages/build-tools/src/common/fluidUtils.ts

```diff
@@ -31,35 +32,72 @@ async function isFluidRootPackage(dir: string) {
 	return false;
 }
 
-async function inferRoot() {
-	let fluidConfig = findUp.sync("fluidBuild.config.cjs", { cwd: process.cwd(), type: "file" });
-	if (fluidConfig === undefined) {
-		traceInit(`No fluidBuild.config.cjs found. Falling back to git root.`);
+async function inferRoot(buildRoot: boolean) {
+	const config = await findUp("fluidBuild.config.cjs", {
+		cwd: process.cwd(),
+		type: "file",
+	});
+	if (config !== undefined) {
+		return path.dirname(config);
+	}
+
+	traceInit(`No fluidBuild.config.cjs found. Falling back to git root.`);
+	try {
 		// Use the git root as a fallback for older branches where the fluidBuild config is still in
 		// package.json
 		const gitRoot = childProcess
-			.execSync("git rev-parse --show-toplevel", { encoding: "utf8" })
+			.execSync("git rev-parse --show-toplevel", {
+				encoding: "utf8",
+				stdio: ["ignore", "pipe", "ignore"],
+			})
 			.trim();
-		fluidConfig = path.join(gitRoot, "package.json");
-		if (fluidConfig === undefined || !existsSync(fluidConfig)) {
-			return undefined;
+
+		const gitRootPackageJson = path.join(gitRoot, "package.json");
+		if (existsSync(gitRootPackageJson)) {
+			if (!buildRoot) {
+				return gitRoot;
+			}
+			// For build root, we require it to have fluidBuild property.
```

Comment:
Why is this?

Reply:
Just didn't want to change behavior for non-fluid build scenarios. But probably can remove it if you think it is fine to change it.